src/design_notes/fn_type_trait_impls.md

```diff
@@ -0,0 +1,257 @@
+# Extending the capabilities of compiler-generated function types
+
+## Background
+
+Both standalone functions and closures have unique compiler-generated types.
+The rest of this document will refer to both categories as simply "function
+types", and will use the phrase "function types without upvars" to refer to
+standalone functions _and_ closures without upvars.
+
+Today, these function types have a small set of capabilities, which are
+exposed via trait implementations and implicit conversions.
+
+- The `Fn`, `FnMut` and `FnOnce` traits are implemented based on the way
+  in which upvars are used.
+
+- `Copy` and `Clone` traits are implemented when all upvars implement the
+  same trait (trivially true for function types without upvars).
+
+- `auto` traits are implemented when all upvars implement the same trait.
+
+- Function types without upvars have an implicit conversion to the
+  corresponding _function pointer_ type.
+
```

Comment:
```suggestion

This design note attempts to capture a range of proposed solutions, and some of the feedback regarding those solutions. This design note does not intend to favor any specific solutions, just reflect past discussions. The presence or absence of any particular feedback in this document does not necessarily serve to favor or disfavor any particular solution.
```

Reply:
I think something like this would help avoid making this feel like an RFC, because as it stands, this almost feels more like part of an RFC than like a design note.